src/main/java/de/tum/in/www1/artemis/service/metis/PostService.java

```diff
@@ -251,6 +246,14 @@ else if (postContextFilter.getCourseWideContext() == null && postContextFilter.g
             throw new BadRequestAlertException("A new post cannot be associated with more than one context", METIS_POST_ENTITY_NAME, "ambiguousContext");
         }
 
+        // sets author roles to display user authority icon on posting headers
+        postsInCourse.forEach(post -> {
+            Course postingCourse = post.getCourse() != null ? post.getCourse()
```

Comment:
Those are three chained `?:` operators which makes this incredibly hard to read. Please do not use the ternary operator more than once in a single statement.

Reply:
thanks